Auditing Resource and Data Source (#82)
mongodb/terraform-provider-mongodbatlas: website/docs/r/auditing.html.markdown

---
layout: "mongodbatlas"
page_title: "MongoDB Atlas: auditing"
sidebar_current: "docs-mongodbatlas-resource-auditing"
description: |-
    Provides a Auditing resource.
---

# mongodbatlas_auditing

> `mongodbatlas_auditing` provides a Auditing resource. This allows auditing to be created.

Reviewer: ```suggestion
`mongodbatlas_auditing` provides an Auditing resource. This allows auditing to be created.
```

## Example Usage

```hcl
resource "mongodbatlas_auditing" "test" {
		project_id                  = "<project-id>"
		audit_filter                = "{ 'atype': 'authenticate', 'param': {   'user': 'auditAdmin',   'db': 'admin',   'mechanism': 'SCRAM-SHA-1' }}"
		audit_authorization_success = false
		enabled                     = true
	}
```

## Argument Reference

> * `project_id` - (Required) The unique ID for the project to create the database user.

Reviewer: ```suggestion
* `project_id` - (Required) The unique ID for the project to configure auditing.
```

> * `audit_filter` - Indicates whether the auditing system captures successful authentication attempts for audit filters using the "atype" : "authCheck" auditing event. For more information, see auditAuthorizationSuccess

Reviewer: Can we actually link `auditAuthorizationSuccess` to the appropriate MongoDB docs page? By itself, it doesn't make much sense here.

Reviewer: ```suggestion
* `audit_filter` - Indicates whether the auditing system captures successful authentication attempts for audit filters using the "atype" : "authCheck" auditing event. For more information, see [auditAuthorizationSuccess](https://docs.mongodb.com/manual/reference/parameters/#param.auditAuthorizationSuccess).
```

* `audit_authorization_success` - JSON-formatted audit filter used by the project

> * `enabled` - Denotes whether or not the project associated with the {GROUP-ID} has database auditing enabled.

Reviewer: We should replace {GROUP-ID} with project ID to be consistent

~> **NOTE:** Auditing created by API Keys must belong to an existing organization.

## Attributes Reference

In addition to all arguments above, the following attributes are exported:

> * `configuration_type` - Denotes the configuration method for the audit filter. Possible values are: NONE - auditing not configured for the project.m FILTER_BUILDER - auditing configured via Atlas UI filter builderm FILTER_JSON - auditing configured via Atlas custom filter or API.

Reviewer: [typo] `m`

Reviewer: Can we markdown format the possible values into a list?
